### PR TITLE
[SW-1737] Ensure the client has full flatfile in external backend

### PR DESCRIPTION
**Pull request by jakubhava**
Depends on https://github.com/h2oai/h2o-3/pull/4111

### CODE REVIEW

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala [hunk 1]
```diff
@@ -65,7 +65,7 @@ trait ExternalBackendConf extends SharedBackendConf {
     PROP_EXTERNAL_COMMUNICATION_BLOCK_SIZE._2)
   def externalCommunicationBlockSize: String = sparkConf.get(PROP_EXTERNAL_COMMUNICATION_BLOCK_SIZE._1, PROP_EXTERNAL_COMMUNICATION_BLOCK_SIZE._2)
   private[backends] def isBackendVersionCheckDisabled() = sparkConf.getBoolean(PROP_EXTERNAL_DISABLE_VERSION_CHECK._1, PROP_EXTERNAL_DISABLE_VERSION_CHECK._2)
-
+  private[backends] def externalBackendFlatfileName() = sparkConf.getOption(PROP_EXTERNAL_FLATFILE_NAME._1)
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: nit: `FlatfileName` -> `FlatFileName`?

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala [hunk 2]
```diff
@@ -233,4 +234,7 @@ object ExternalBackendConf {
 
   /** Disable version check of external H2O backend */
   val PROP_EXTERNAL_DISABLE_VERSION_CHECK = ("spark.ext.h2o.external.disable.version.check", false)
+
+  /** Name of file where flatfile is stored in case of automatic mode of external backend */
+  val PROP_EXTERNAL_FLATFILE_NAME = ("spark.ext.h2o.external.flatfile.name", None)
```
Comment (mn-mikke) [inline core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: Just a question: Doesn't the preporty describe a relative path of a flat file and not just name?

Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: Nope, just a name

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala (hunk 1 above)
Comment (jakubhava) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: 👍

File: core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala (hunk 2 above)
Comment (mn-mikke) [reply core/src/main/scala/org/apache/spark/h2o/backends/external/ExternalBackendConf.scala]: Ok, thanks!

**Pull request merged**